react/features/recent-list/reducer.js

```diff
@@ -30,7 +24,9 @@ PersistencyRegistry.register(STORE_NAME, {
 /**
  * Reduces the Redux actions of the feature features/recent-list.
  */
-ReducerRegistry.register(STORE_NAME, (state = DEFAULT_STATE, action) => {
+ReducerRegistry.register(STORE_NAME, (state = {
+    list: getLegacyRecentRoomList()
```

Comment:
I think here's an implicit dependency that by the time the first redux action gets dispatched the implementation of window.localStorage will have finished loading (given that it may be asynchronous like AsyncStorage). Is the reliance really necessary?

Reply:
B. I think it'd be cleaner to remove the legacy value from `window.localStorage` after we've successfully transitioned it to the new format/key.